### PR TITLE
ADR 250: add minimap specs 

### PR DESCRIPTION
**Pull request by leanmendoza**
Check https://github.com/decentraland/protocol/pull/206

### CODE REVIEW

File: content/ADR-250-minimap-standard.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+  "parcelView": {
+    "version": "v1",
+    "imageUrl": "https://api.decentraland.org/v1/minimap.png"
+ }
+}
+```
+
+
+## size
+- The union of all rectangles here represents the places where can have scenes
```
Comment (marianogoldman) [inline content/ADR-250-minimap-standard.md:77]:
```suggestion
- The union of all rectangles here represents the places where that have scenes
```

File: content/ADR-250-minimap-standard.md [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+- All the other parcels are considered empty; it is up to the explorer to decide if they are walkable or not
+
+With these parameters we can compute an important square, which is the contained one, and the following vertices determine it:
+  - top-left = (minX, maxY) 
+  - bottom-left = (minX, minX) 
+  - top-right = (maxX, maxY) 
+  - bottom-right = (maxX, maxY)
```
Comment (marianogoldman) [inline content/ADR-250-minimap-standard.md:84]:
```suggestion
  - bottom-right = (maxX, minY) 
```

File: content/ADR-250-minimap-standard.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+```
+
+
+## size
+- The union of all rectangles here represents the places where can have scenes
+- All the other parcels are considered empty; it is up to the explorer to decide if they are walkable or not
+
+With these parameters we can compute an important square, which is the contained one, and the following vertices determine it:
+  - top-left = (minX, maxY) 
+  - bottom-left = (minX, minX)
```
Comment (marianogoldman) [inline content/ADR-250-minimap-standard.md:82]:
```suggestion
  - bottom-left = (minX, minY) 
```

File: content/ADR-250-minimap-standard.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+|2   |6.4         | 64x64 |
+|3   |12.8        | 128x128 |
+|4   |25.6        | 256x256 |
+|5   |102.4       | 1024x1024 |
+|6   |204.8       | 2048x2048 |
+
+It allows the Explorers at least to render a basic shape to Explore every Realm.
+
+## Parcel view
+This option is based on the also implemented in the reference client with colored parcels. Due to the little information needed for each parcel, an RGB pixel is an oversized amount of data. However, a simple and extensible solution is that each pixel represents each parcel.
```
Comment (marianogoldman) [inline content/ADR-250-minimap-standard.md:113]:
```suggestion
This option is based on what is implemented in the reference client with colored parcels. Due to the little information needed for each parcel, an RGB pixel is an oversized amount of data. However, a simple and extensible solution is that each pixel represents each parcel. 
```

File: content/ADR-250-minimap-standard.md [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+ vec4 pixel_data = texelFetch(map_data, ivec2(int(cx), int(cy)), 0);
+ int flagsR = int(pixel_data.r * 255.0);
+ int flagsG = int(pixel_data.g * 255.0);
+
+ bool topMask = (flagsR & 0x8) > 0;
+ bool leftMask = (flagsR & 0x10) > 0;
+  
+ vec3 parcel_color;
+ if (flagsG == 32) {
+ parcel_color = colors[1];
```
Comment (marianogoldman) [inline content/ADR-250-minimap-standard.md:176]: Please fix indentation here

File: content/ADR-250-minimap-standard.md [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+ if (selected_tile.x == cx && selected_tile.y == cy) {
+ resolved_color += vec4(0.7, 0.1, 0.1, COLOR.a);
+ }
+  
+ COLOR = resolved_color;
+}
+```
+
+# Conclusion
+This is an optional feature that Realm can use to specify the explorer how to render the Minimap or not. The example for the Genesis-City is straightforward and ready to be implemented in the `/about` request.
```
Comment (marianogoldman) [inline content/ADR-250-minimap-standard.md:222]:
```suggestion
This is an optional feature that Realm can use to specify the Explorer how to render the Minimap (or instruct not to render it). The example for the Genesis City is straight-forward and ready to be implemented in the `/about` request.
```

**Review by marianogoldman — CHANGES_REQUESTED** [review]

**Review by kuruk-mm — APPROVED** [review]
Looks great!

File: content/ADR-250-minimap-standard.md (hunk 2 above)
Comment (leanmendoza) [reply content/ADR-250-minimap-standard.md:85]: I left a note comment, it's a different coordinate system

**Review by kuruk-mm — APPROVED** [review]

**Review by marianogoldman — APPROVED** [review]

**Pull request merged**